frontend/src/metabase/core/components/TabRow/TabRow.tsx

```diff
@@ -38,14 +58,36 @@ function TabRowInner<T>({
     tabListRef.current.scrollBy(scrollDistance, 0);
   };
 
+  const pointerSensor = useSensor(PointerSensor, {
+    activationConstraint: { distance: 10 },
+  });
+
+  const onDragEnd = (event: DragEndEvent) => {
+    if (!event.over || !handleDragEnd) {
+      return;
+    }
+    handleDragEnd(event.active.id, event.over.id);
+  };
+
   return (
     <TabList
       onChange={onChange as (value: unknown) => void}
       onScroll={event => setScrollPosition(event.currentTarget.scrollLeft)}
       ref={tabListRef}
       {...props}
     >
-      {children}
+      <DndContext
```

Comment:
See [dnd-kit `sortable` docs](https://docs.dndkit.com/presets/sortable) for context on what's being added here.